python/grass/pygrass/gis/__init__.py

```diff
@@ -440,15 +440,14 @@ def __iter__(self):
 
     def read(self):
         """Return the mapsets in the search path"""
-        with open(self.spath, "rb") as f:
-            lines = f.readlines()
-            if lines:
-                return [decode(line.strip()) for line in lines]
-        lns = [
-            "PERMANENT",
-        ]
-        self._write(lns)
-        return lns
+        try:
+            with open(self.spath, "rb") as f:
```

Comment:
`open` file func `mode` param arg should be `"r"` only according output bellow and we don't need decode readed lines:

```
GRASS nc_spm_08_grass7/landsat:~ > g.mapsets operation=set mapset=landsat

GRASS nc_spm_08_grass7/landsat:~ > file grassdata/nc_spm_08_grass7/landsat/SEARCH_PATH 
grassdata/nc_spm_08_grass7/landsat/SEARCH_PATH: ASCII text
```

Same for `_write()` method, `open` file func with  `mode` param arg should be `w` only.

```python
    def _write(self, mapsets):
        """Write to SEARCH_PATH file the changes in the search path

        :param list mapsets: a list of mapset's names
        """
        with open(self.spath, "w") as f:
            f.write("\n".join([m for m in mapsets if m in self.location.mapsets()]))
```

`extend()` method without `decode` func

```python
    def extend(self, mapsets):
        """Add more mapsets to the search path

        :param list mapsets: a list of mapset's names
        """
        final.extend(
            [
                m
                for m in mapsets
                if m in self.location.mapsets() and m not in self.read()
            ]
        )
        self._write(final)
```